### DISCUSSION TITLE
Constraints on relations to same collection

### DISCUSSION THREAD

**Discussion by gerolds** [directus/directus#6736] [Feature Requests]
We create graph-like structures by adding a Many-To-One field to a collection that points to itself (e.g. parent, successor etc.). Currently, without constraints this allows us (anyone) to create any type of graph structure, including one that include cycles.

To create more constrained structures like trees, linked lists, rings we would need to constrain which items in a collection can be selected in the relation, i.e. "disallow self, disallow descendants, disallow ancestor, ...".

Such constraints can be manually created (i.e. excluding specific items via roles) but this is tedious and brittle.

A desireable feature would be an configurable constraint (allow/disallow <node-property>) on the collection field that holds the foreign key.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
